### ISSUE TITLE
E303: missing detection

### ISSUE THREAD

**Issue by spaceone** [astral-sh/ruff#10374]
E303 not detected for:

```python
#



import foo


def bar() -> list[str]:
    pass
```

flake8 detects: `:5:1: E303 too many blank lines (3)`

**Comment by boolean-light** [comment]
Probably it is in `crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs`?

**Comment by dhruvmanila** [comment]
Thanks for the report.

From the implementation side, I guess this must be a bug in the order in which the check is performed:

https://github.com/astral-sh/ruff/blob/dacec7377c436fc85e380f7cc69a007824fa08bc/crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs#L699-L701

and the state is being updated:

https://github.com/astral-sh/ruff/blob/dacec7377c436fc85e380f7cc69a007824fa08bc/crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs#L735-L736

/cc @hoel-bagard

**Comment by hoel-bagard** [comment]
I'll have a look at it, thanks for reporting the bug.
